packages/driver/test/client.test.ts

```diff
@@ -2602,6 +2602,42 @@ if (getEdgeDBVersion().major >= 6) {
     }
   });
 
+  test("querySQL codec", async () => {
+    let client = getClient();
+
+    try {
+      let res = await client.querySQL("select 1");
+      expect(JSON.stringify(res)).toEqual("[[1]]");
+
+      res = await client
+        .withCodecs({
+          sql_row: {
+            fromDatabase(data, desc) {
+              return Object.fromEntries(
+                desc.names.map((key, index) => [key, data[index]]),
+              );
+            },
+            toDatabase() {
+              // toDatabase is required to be specified (limitation
+              // of TypeScript type system). This isn't super elegant
+              // and most likely we'll need a higher-level nicer API
+              // in top of this.
+              //
+              // Maybe `client.withCodecs(gel.SQLRowAsObject)`?
```

Comment:
Yes, absolutely, 100%. Could even short-cut this with `.withSQLObjectCodec()`. I almost think that _this_ should be the default and have the array of tuples be the opt-in option.

Reply:
OK, let's do this. I'll push some updates in a few.